includes/class-omise-callback.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 4.0
+ */
+class Omise_Callback {
+	/**
+	 * @var \WC_Abstract_Order
+	 */
+	protected $order;
+
+	/**
+	 * @var \OmiseCharge
+	 */
+	protected $charge;
+
+	/**
+	 * @param \WC_Abstract_Order $order
+	 */
+	public function __construct( $order ) {
+		$this->order = $order;
+		if ( ! $this->order || ! $this->order instanceof WC_Abstract_Order ) $this->invalid_result();
+	}
+
+	public static function execute() {
+		$order_id = isset( $_GET['order_id'] ) ? sanitize_text_field( $_GET['order_id'] ) : null;
+
+		$callback = new self( wc_get_order( $order_id ) );
+		$callback->validate();
+	}
+
+	public function validate() {
+		$this->order->add_order_note( __( 'OMISE: Validating the payment result...', 'omise' ) );
+
+		try {
+			$this->charge = OmiseCharge::retrieve( $this->order->get_transaction_id() );
+
+			switch ( strtolower( $this->charge['status'] ) ) {
+				case 'successful':
```

Comment:
I see you 👍'd my previous comment, but dropping this here again:

https://github.com/omise/omise-woocommerce/pull/167/files/e444e4a05d89bfa092d3d06dd754fdad8a33a568#r439242588

Reply:
@danfowler thanks, I left those 2 statuses out as there is no particular case to handle at the moment. But noted for a future improve 👍